web/modules/custom/dept_node/dept_node.module

```diff
@@ -267,9 +272,63 @@ function dept_node_assign_domain_defaults(array $form, FormStateInterface $form_
   $domain_access[] = ['target_id' => 'nigov'];
 
   // Remove any duplicate domain entries.
-  array_unique($domain_access, SORT_REGULAR);
+  $domain_access = array_unique($domain_access, SORT_REGULAR);
 
   $form_state->setValue('field_domain_access', $domain_access);
+
+  dept_node_assign_domain_defaults_media($form, $form_state, $domain_access);
+}
+
+/**
+ * Internal callback to set matching domain_access values on media entities
+ * referenced from press release news nodes. This ensures both the news node
+ * and referenced media entities can appear on both the source domain as well
+ * as the nigov domain.
+ */
+function dept_node_assign_domain_defaults_media(array $form, FormStateInterface $form_state, array $domain_access) {
+  // @phpstan-ignore-next-line
+  $bundle = $form_state->getFormObject()->getEntity()->bundle();
+
+  // Replicate the domain access settings on any referenced media entities.
+  $field_definitions = \Drupal::service('entity_field.manager')->getFieldDefinitions('node', $bundle);
+
+  // Initialize an array to store media entity reference fields.
+  $media_reference_fields = [];
+
+  // Loop through field definitions to find media entity reference fields.
+  foreach ($field_definitions as $field_name => $field_definition) {
+    if ($field_definition instanceof BaseFieldDefinition) {
+      // Skip base fields (e.g., 'title', 'status').
+      continue;
+    }
+
+    $field_storage = FieldStorageConfig::loadByName('node', $field_name);
+
+    // Check if the field is a reference field and targets media entities.
+    if ($field_storage && $field_storage->getSetting('target_type') === 'media') {
+      $media_reference_fields[] = $field_name;
+    }
+  }
+
+  if (!empty($media_reference_fields)) {
+    foreach ($media_reference_fields as $field_name) {
+      $field_value = $form_state->getValue($field_name);
+
+      if (empty($field_value['selection'])) {
+        continue;
+      }
+
+      // Load the media entity value and set the domain access values for it.
+      foreach ($field_value['selection'] as $key => $value) {
+        $media = Media::load($value['target_id']);
+
+        if ($media instanceof MediaInterface) {
+          $media->set('field_domain_access', $domain_access);
```

Comment:
If it's an existing media entity and has access set on another dept to the current this will overwrite that. Do we need to append this dept to any existing values?